docs/SECURITY.md

```diff
@@ -0,0 +1,81 @@
+
+## EVE deployment model
+
+Reasoning about security and trustworthiness of any system is only meaningful if one understands how the systems is going to be deployed and used. EVE main deployment target are Edge Nodes. Typically, Edge Nodes are small form factor, ruggedized PC-like systems that are deployed in physically insecure environments with ad-hoc networking without any firewall-like guarantees. This type of environment is very similar to the one assumed by mobile computing systems (laptops, tablets and smart phones) and it is not a coincidence that EVE shares a lot of security principles with operating systems developed for mobile computing. For example, EVE's approach to thwarting physical attacks and protecting the integrity of the software stack running on the Edge Node may look similar to if you're familiar with Chromium OS.
+
+There is, however, one crucial difference between mobile computing environment and Edge and that is the notion of who controls the system. With mobile there's always an human being end user who has direct physical access to the hardware running mobile computing operating system, in EVE's case that entity is a remote, cloud-based controller. Any security related event that originates on EVE's side (e.g. measurements coming from boot sequence) has to be reliably passed and acknowledged by the controller and any action that is triggered on the Edge Node by EVE (e.g. storage key rotation) has to be scheduled by the controller.
+
+In short, EVE's deployment model forces us to protect against an opportunistic adversary with full physical access to and Edge Node and Edge Node's networks through a combination of system hardening, defense in depth, process isolation, secure autoupdate, measured boot, encryption, and seamless integration with EVE's controller security policies.
+
+## Security capabilities vs. security policies
+
+At the end of the day, EVE only does what its controller instructs it to do. This puts an enormous premium on protecting the trust between EVE instance and its controller at all costs (in both direction) but it also allows us to have a clear division of responsibilities between the two:
+
+* EVE only implements security capabilities, never security policies
+* Controller orchestrates EVE's security capabilities in order to realize sophisticated policies
+
+On the EVE's side, this separation allows us to focus on a small number of well designed capabilities (AKA security building blocks) that are guaranteed to have a well defined interactions between each other. For example, while EVE may provide a capability for storing encrypted binary blobs on the Edge Nodes it will be up to the controller's security policy to determine which pieces of data will be stored there.
+
+Controller and its management of security policies are out of scope for this document.
+
+## EVE threat model
+
+Borrowing a great deal from Chromium effort we consider two different kinds of adversaries:
```

Comment:
Recommendation: I don't think it adds to write "Borrowing ... from Chromium effort." We can just say, "We consider two different kinds of adversaries:" We are good enough to speak ourselves. :-)

Reply:
Sure, I suppose I reference them in the beginning anyway ;-)